### ISSUE TITLE
Windows/TUI: Text overrides line numbers

### ISSUE THREAD

**Issue by kerrytazi** [neovim/neovim#9461]
- `nvim --version`: NVIM v0.3.3 Build type: RelWithDebInfo
- Operating system/version: Windows 10 Pro x64 latest
- Terminal name/version: cmd & ConEmu latest

Add or delete any line and buffer overrides line numbers. `Ctrl + L` helps but it's annoying.

### Steps to reproduce using `nvim -u NORC`

```
nvim -u NORC
:set nu
:edit main.cpp
dd
```
Example file: main.cpp
```
#include <string>
#include <regex>


void via_regex()
{
	std::string content("4aejuj+8tr.v,ebyv-67.3yt3ktux+3.3");

	std::regex re("[+-]?\\d+\\.\\d+");

	std::sregex_iterator begin(content.cbegin(), content.cend(), re);
	std::sregex_iterator end;

	for (auto it = begin; it != end; ++it)
	{
		print_out("regex result: ", it->str(), '\n');
	}
}
```

### Actual behavior
![screenshot](https://imgur.com/WkVo7IR.png)

### Expected behaviour
![screenshot](https://imgur.com/YtLu8fS.png)

**Comment by justinmk** [comment]
- what does `:echo &term` say ?
- please check if the problem happens with the .zip from https://github.com/neovim/neovim/pull/9094 : [64 bit](https://ci.appveyor.com/project/neovim/neovim/builds/21399882/job/okjaemai392ohxv8/artifacts) or [32 bit](https://ci.appveyor.com/project/neovim/neovim/builds/21399882/job/bgy9tisjdf844vf5/artifacts)

**Comment by justinmk** [comment]
can you please try the .zip files I mentioned above.

**Comment by kerrytazi** [comment]
`:echo &term`
`builtin_vtpcon`

`:echo $TERM`
`vtpcon`

Just tried nvim from your link. Problem still exists.

**Comment by justinmk** [comment]
> Just tried nvim from your link. Problem still exists.

Thanks, so at least this wasn't a regression that I caused when merging parts of #9094. 

@erw7 have you seen this issue before?

**Comment by erw7** [comment]
> have you seen this issue before?

I will see this problem for the first time. But, I saw a similar problem when I used `vs:` command.

+ nvim --version: NVIM v0.4.0-55-g5fba81592 Build type: RelWithDebInfo
+ Operating system/version: Microsoft Windows [Version 6.1.7601]
+ Terminal name/version: ConEmu 180626 [64]
+ $TERM: conemu
+ &term: builtin_conemu

### Steps to reproduce using nvim -u NORC
```
nvim -u NORC
:vs
:e src/nvim/tui/tui.c
<C-w>l
:e src/nvim/tui/input.c
<C-f>
```

In this operation, some lines override rows in the right window. Should I create a new issue?

In the case of `set TERM=win32con`, this problem does not seem to happen.

**Comment by krishnakumarg1984** [comment]
@justinmk No, using that zip file does not solve the problem. I am recently forced to migrate to windows due to recent change of jobs (corporate IT won't install *nix for me), so I can't comment whether this is a regression. I have never seen this on a Mac or linux, and as you can imagine this behaviour of nvim is quite annoying.

**Comment by justinmk** [comment]
@kerrytazi @krishnakumarg1984 can you try the artifact (Neovim.zip) from #9474:

- [64 bit](https://ci.appveyor.com/project/neovim/neovim/builds/21460739/job/sloe48vw25wt3rui/artifacts)
- [32 bit](https://ci.appveyor.com/project/neovim/neovim/builds/21460739/job/n8m9kdec8pu8yd72/artifacts)

**Comment by krishnakumarg1984** [comment]
@justinmk  yes, #9474 works perfect for me too.

**Comment by BYK** [comment]
@justinmk I'm still having this issue when using Alacritty on Windows. How can I make neovim to send cursor down instead of a new line on Alacritty too?

**Comment by justinmk** [comment]
@BYK always need to know exact Nvim version

**Comment by BYK** [comment]
@justinmk I am using the latest nightly `v0.4.0-1356-gafef97326` (just dowloaded). I'm also looking at Alacritty's code in case this is a bug there.

**Comment by krishnakumarg1984** [comment]
@justink this bug is persistent even on `cmder`/`conemu` on Windows, using `nvim 0.4-dev`

**Comment by BYK** [comment]
@justinmk so here are my findings:
 - My `$TERM` is `xterm-256color`
 - In trace mode, alacritty reports receiving `CR` `LF` sequences during page updates
 - `:checkhealth` reports the following about my terminal:
   ```
   ## terminal
   - INFO: key_backspace (kbs) terminfo entry: key_backspace=^H
   - INFO: key_dc (kdch1) terminfo entry: key_dc=\E[3~
   - INFO: $COLORTERM='truecolor'
   ```

**Comment by erw7** [comment]
@BYK Can the problem be fixed with the #10558 artifacts?

**Comment by erw7** [comment]
@krishnakumarg1984  I also wrote this https://github.com/neovim/neovim/issues/9596#issuecomment-462217801. Did you set the `TERM` environment variable?

**Comment by krishnakumarg1984** [comment]
@erw7  No. `$TERM$  has not been set.

**Comment by justinmk** [comment]
@krishnakumarg1984 please try the artifact from #10558 : https://ci.appveyor.com/project/neovim/neovim/build/job/7guodosue1p6fok8/artifacts

**Comment by erw7** [comment]
@krishnakumarg1984 cmder sets `TERM=cygwin`. For now, please set `TERM=` before starting `neovim`.

Ref. https://github.com/cmderdev/cmder/blob/0280593d4ce8e0f846f26698550907a4b2a4df0e/vendor/init.bat#L161

**Comment by BYK** [comment]
@justinmk sorry for the delay. I couldn't test the artifacts as the executables don't start:

![image](https://user-images.githubusercontent.com/126780/61659292-d8fa4980-accf-11e9-9b1b-bd6fd59787aa.png)

(mingw64 builds as the msvc64 builts were missing)

**Comment by justinmk** [comment]
Are you using the full .zip or only the . exe ?

**Comment by BYK** [comment]
@justinmk tried both, same result. The nightly builds work w/o issues (using the win64 zip).

**Comment by erw7** [comment]
@BYK The error occurs because `libwinpthread-1.dll`(64bit) is missing. I think that you can execute it if you extract `libwinpthread-1.dll` from the [package of MSYS2/MinGW64](http://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-libwinpthread-git-7.0.0.5480.e14d23be-1-any.pkg.tar.xz) to somewhere in the search `PATH`.

**Comment by BYK** [comment]
@erw7 interesting. The Neovim package from the PR already had that DLL in it but replacing it from the one you linked fixed the issue, thanks!

@justinmk I can confirm that the patch solves the issue. 🎉

**Comment by blueyed** [comment]
@erw7 
> The error occurs because `libwinpthread-1.dll`(64bit) is missing.

Is this because using an artifact that is gcov enabled, or not related to the missing dlls we've seen there on AppVeyor at all?  (just wondered)

**Comment by erw7** [comment]
> Is this because using an artifact that is gcov enabled, or not related to the missing dlls we've seen there on AppVeyor at all? (just wondered)

This is the same problem we encountered at #10404. The `libwinpthread-1.dll` included in the package copied from `.dep/usr/bin` comes from the `neovim-qt` package. Currently, only 32 bits of `neovim-qt` package are created. The artifacts that is gcov enabled are 64 bit and will not work with `libwinpthread-1.dll`(32bit) included in the package.
